src/main.py

```diff
@@ -159,15 +164,79 @@ def main():
             os.path.expanduser(args.assistant_secrets))
         recognizer = speech.AssistantSpeechRequest(credentials)
 
-    recorder = audio.Recorder(
-        input_device=args.input_device, channels=1,
-        bytes_per_sample=speech.AUDIO_SAMPLE_SIZE,
-        sample_rate_hz=speech.AUDIO_SAMPLE_RATE_HZ)
-    with recorder:
-        do_recognition(args, recorder, recognizer, player)
+    status_ui = StatusUi(player, args.led_fifo, args.trigger_sound)
 
+    # The ok-google trigger is handled with the Assistant Library, so we need
+    # to catch this case early.
+    if args.trigger == 'ok-google':
```

Comment:
It'd be nice to also make the button work with the library (it supports `start_conversation`).